test/unit-tests/analytics-service.ts

```diff
@@ -511,37 +513,41 @@ describe("analytics-service", () => {
 	describe("uses correct settings on different os-es", () => {
 		let name = "unitTests";
 		let testInjector: IInjector;
-		let osType = os.type;
-		let osRelease = os.release;
+		let osInfo: IOsInfo;
+		let osType: () => string;
+		let osRelease: () => string;
 		let release = "1.0";
 
 		beforeEach(() => {
 			testInjector = createTestInjector(baseTestScenario);
 			service = testInjector.resolve("analyticsService");
+			osInfo = testInjector.resolve("osInfo");
+			osType = osInfo.type;
+			osRelease = osInfo.release;
 		});
 
 		after(() => {
-			os.type = osType;
-			os.release = osRelease;
+			osInfo.type = osType;
```

Comment:
the `after` task should return everything in the correct way, so other tests will use the correct osInfo. In this case we set the `type` property of osInfo to something that is not the default value. Is this correct?